plugins/catalog-backend-module-aws/src/providers/AwsS3EntityProviderFactory.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TaskRunner } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import { InputError } from '@backstage/errors';
+import { AwsS3Integration, ScmIntegrations } from '@backstage/integration';
+import { AwsS3EntityProvider } from './AwsS3EntityProvider';
+import { readAwsS3Configs } from './config';
+import { AwsS3Config } from './types';
+import { Logger } from 'winston';
+
+const AMAZON_AWS_HOST = 'amazonaws.com';
+
+/**
+ * Creates an {@link AwsS3EntityProvider} instance for each provider config.
+ *
+ * @public
+ */
+export class AwsS3EntityProviderFactory {
```

Comment:
We don't typically have separate factory classes, unless you're actually creating a factory. I think `fromConfig` could go on the implementation class instead

Reply:
I tried to reduce the dependencies and complexity a bit, but I guess I can find a way for both more or less. Get your point.

Reply:
I moved it to the provider class and removed the factory.